trade-finance-manager-api/src/v1/controllers/update-facilities.js

```diff
@@ -19,39 +19,41 @@ const updateFacilities = async (deal) => {
 
   modifiedDeal.facilities = await Promise.all(modifiedDeal.facilities.map(async (f) => {
     const facility = f;
-
     const { _id: facilityId } = facility;
+    let facilityUpdate;
+    let facilityPremiumSchedule;
+    let feeRecord;
 
     const facilityGuaranteeDates = getGuaranteeDates(facility, dealSubmissionDate);
-
     const facilityCurrencyConversion = await convertFacilityCurrency(facility, dealSubmissionDate);
     const facilityExposurePeriod = await getFacilityExposurePeriod(facility);
 
-    let facilityPremiumSchedule;
+    // Premium Schedule is only valid for non-GEF facilities
     if (dealType === CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS) {
       facilityPremiumSchedule = await getFacilityPremiumSchedule(
         facility,
         facilityExposurePeriod,
         facilityGuaranteeDates,
       );
+      facilityUpdate = {
+        premiumSchedule: facilityPremiumSchedule,
+      };
+    } else if (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF) {
+      // Fee record is only valid for GEF facilities
+      if (submissionType !== CONSTANTS.DEALS.SUBMISSION_TYPE.MIA) {
```

Comment:
Can we combine the `else if` and `if` ?


```
else if (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF
         && submissionType !== CONSTANTS.DEALS.SUBMISSION_TYPE.MIA) { ....
```

Reply:
My intention was to make this future proof in case we have any other records or computation to be done purely for GEF.

Reply:
makes sense 👍